### PR TITLE
feat(Packaging): Build and release deb and rpm packages

### PR DESCRIPTION
**Pull request by jsirianni**
## Description of Changes

Added deb and rpm package building to Goreleaser.

Set binary name to `stanza` and disabled `no_unique_dist_dir` as this is the name that will be used during package install (`/usr/bin/stanza`).

Package install will
- Add stanza binary to /usr/bin/stanza, this is a slight divergence from how we traditionally install stanza.
- Add plugins (already managed by Goreleaser)
- Add default config (config was taken from install script)
- Run pre install script: Configure `stanza` user
- Run post install script: Handles service creation and final permissions

Added CINC Test Kitchen config
- Wraps vagrant for easy testing
- As simple as `kitchen test debian-10`
- Ensures Stanza was installed correctly

Documented how to use CINC / Test Kitchen.
Documented package install process.
Documented how to use a root user when installing with a package manager.

## Questions

On release, I believe the binary file names will be
```
stanza_linux_arm64_1.6.0_linux_arm64
stanza_windows_amd64_1.6.0_windows_amd64.exe
stanza_darwin_amd64_1.6.0_darwin_amd64
stanza_linux_amd64_1.6.0_linux_amd64
stanza_darwin_arm64_1.6.0_darwin_arm64
```

Which is not going to work with the current install script (line 309):
```
...
  if [ -z "$version" ] ; then
    agent_download_url="$url/latest/download/${BINARY_NAME}_${os}_${os_arch}"
    plugins_download_url="$url/latest/download/${PLUGINS_PACKAGE}"
  else
    agent_download_url="$url/download/v$version/${BINARY_NAME}_${os}_${os_arch}"
    plugins_download_url="$url/download/v$version/${PLUGINS_PACKAGE}"
  fi
...
```

Should that be addressed with this PR?

## **Please check that the PR fulfills these requirements**
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)
- [x] CI passes

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -16,4 +16,4 @@ gen/
 **/wix.dynamic.json
 stanza-plugins/
 **/plugins
-
+**/.kitchen
```
Comment (cpheps) [inline .gitignore:19]: newline

File: build/package/config.yaml [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  #   credentials_file: /my/credentials/file
+
+  # An example output that sends captured logs to elasticsearch.
+  # For more info: https://github.com/observIQ/stanza/blob/master/docs/operators/elastic_output.md
+  #
+  # - id: example_output
+  #   type: elastic_output
+  #   addresses:
+  #     - http://my_node_address:9200
+  #   api_key: my_api_key
```
Comment (cpheps) [inline build/package/config.yaml:35]: newline

File: docs/development.md [hunk 3]
```diff
@@ -24,6 +24,32 @@ make build-darwin-amd64
 
 Tests can be run with `make test`.
 
+## Integration Tests with CINC Test Kitchen
```
Comment (cpheps) [inline docs/development.md:27]: What is the purpose of the CINC stuff? I've not seen this before do we have a standard for using it elsewhere?

File: build/post.sh [hunk 4]
```diff
@@ -18,3 +18,5 @@ fi
 if [ ! -f ./artifacts/windows-install.ps1 ]; then
     cp ./scripts/windows-install.ps1 ./artifacts/windows-install.ps1
 fi
+
+tar -xf ./artifacts/stanza-plugins.tar.gz -C ./artifacts
```
Comment (cpheps) [inline build/post.sh:22]: newline

File: .goreleaser.yml [hunk 5]
```diff
@@ -29,6 +29,26 @@ archives:
...
+  - package_name: stanza
+    vendor: observIQ, Inc
+    maintainer: observIQ <support@observiq.com>
+    license: Apache 2.0
+    bindir: /usr/bin
+    formats:
+      - rpm
+      - deb
+    contents:
+      - src: artifacts/plugins
```
Comment (cpheps) [inline .goreleaser.yml:42]: Maybe I'm missing it but how do the plugins get into the artifacts folder?

File: docs/development.md (hunk 3 above)
Comment (jsirianni) [reply docs/development.md:27]: Everytime. I swear I went through and checked :upside_down_face:

File: build/package/config.yaml (hunk 2 above)
Comment (jsirianni) [reply build/package/config.yaml]: Everytime. I swear I went through and checked :upside_down_face:

File: docs/development.md (hunk 3 above)
Comment (jsirianni) [reply docs/development.md:27]: CINC is an open source fork of Chef. Chef's license changed as is not suitable for us. I use CINC's Auditor (chef inspec) to validate post install state.

Comment (cpheps) [reply docs/development.md:27]: I agree testing is worth while. I'm ok with this for now but my fears are:
- bloat in our repos (maybe we move this tooling to an internal repo that just runs these types of tests)
- Complex tooling that doesn't work everywhere. 

As an action item can you write up a proposal for using this or submit something to the standards repo if we want to move forward with this? I'm ok with Stanza being the poc repo for this but I would like some discussion before we put this heavy tooling in everywhere.

**Review by cpheps — APPROVED** [review]

File: docs/development.md (hunk 3 above)
Comment (jsirianni) [reply docs/development.md:27]: Sounds good :+1:

File: .goreleaser.yml [hunk 6]
```diff
@@ -19,7 +19,7 @@ builds:
     ldflags:
       - -X github.com/observiq/stanza/version.GitTag={{ .Tag }}
       - -X github.com/observiq/stanza/version.GitCommit={{ .FullCommit }}
-    no_unique_dist_dir: true
+    no_unique_dist_dir: false
```
Comment (BinaryFissionGames) [inline .goreleaser.yml:22]: Do we know how this affects our releases?

Comment (BinaryFissionGames) [reply .goreleaser.yml:22]: my assumption is it still just attaches the binaries, just want to confirm. There are already some problems with the naming that I'm trying to fix here https://github.com/observIQ/stanza/pull/591

Comment (jsirianni) [reply .goreleaser.yml:22]: Goreleaser keeps track of their location on disk and will upload them the same way. I suspect the naming is wrong (and has been wrong), so I am ignoring that here. Added a comment to your PR.

**Review by BinaryFissionGames — APPROVED** [review]

**Pull request merged**